OreshkoJA/lab_1/lab_1.cpp

```diff
@@ -0,0 +1,177 @@
+#include <iostream>
+#include <memory.h>
+using namespace std;
+using T = int;
+class Vector {
+public:
+	Vector();
+	~Vector();
+	Vector(unsigned size);
+	Vector(unsigned size, const T& val);
+	unsigned size();
+	void resize(unsigned size);
+	bool insert(unsigned pos, const T& val);
+	void push_back(const T& val);
+	bool contains(const T& val);
+	T & operator [](unsigned index);
+	int find(const T& val);
+	T* data();
+	Vector&operator = (const Vector& vec);
+	Vector(const Vector&vec);
+private:
+	unsigned Size;
+	T*mass;
+};
+Vector::Vector()
+{
+	Size = 0;
+	mass = nullptr;
+};
+Vector::~Vector()
+{
+	Size = 0;
+	delete[]mass;
+};
+Vector::Vector(unsigned size)
+{
+	Size = size;
+	mass = new T[Size];
+	for (int i = 0; i < Size; ++i)
+		mass[i] = T();
+};
+Vector::Vector(unsigned size, const T& val)
+{
+	Size = size;
+	mass = new T[Size];
+	for (int i = 0; i < Size; ++i)
+		mass[i] = val;
+}
+unsigned Vector::size()
+{
+	return Size;
+}
+void Vector::resize(unsigned size)
+{
+	T*old_mass = mass;
+	mass = new T[size];
+	if (size > Size)
+	{
+		memcpy(mass, old_mass, sizeof(T)*Size);
+		for (int i = Size; i < size; ++i)
+			mass[i] = 0;
+	}
+	else
+		memcpy(mass, old_mass, sizeof(T)*size);
+	delete[]old_mass;
+	Size = size;
+}
+bool Vector::insert(unsigned pos, const T& val)
+{
+	if (pos <= Size)
+	{
+		resize(Size + 1);
+		for (int i = Size-1; i > pos; --i)
+			mass[i] = mass[i - 1];
+		mass[pos-1] = val;
+		return true;
+	}
+	else
+		return false;
+}
+void Vector::push_back(const T& val)
+{
+	resize(Size + 1);
+	mass[Size-1] = val;
+}
+bool Vector::contains(const T& val)
+{
+	bool prov = false;
+	for (int i = 0; i < Size; ++i)
+	{
+		if (mass[i] == val)
+			prov = true;
+	}
+	return prov;
+}
+T & Vector::operator [](unsigned index)
+{
+	return mass[index];
+}
+int Vector::find(const T& val)
+{
+	int index = -1;
+	for (int i = 0; i<Size; ++i)
+	{
+		if (mass[i]==val)
+			index = i;
+	}
+	return index;
+}
+T* Vector::data()
+{
+	return mass;
+}
+Vector&Vector::operator =(const Vector&vec)
+{
+	if (this != &vec) 

```

Comment:
Non-void function should have return statement in all possible execution branches